common/rpc/encryption/tlsFactory.go

```diff
@@ -44,38 +44,27 @@ type (
 	CertProvider interface {
 		FetchServerCertificate() (*tls.Certificate, error)
 		FetchClientCAs() (*x509.CertPool, error)
-		FetchServerRootCAsForClient() (*x509.CertPool, error)
 		GetSettings() *config.GroupTLS
 	}
 
+	// ClientCertProvider is an interface to load raw TLS/X509 primitives for configuring clients.
+	ClientCertProvider interface {
+		FetchClientCertificate() (*tls.Certificate, error)
```

Comment:
[Nit] FetchClientCertificate() seems a bit too ambiguous of a name given that it is on a ClientCertProvider. Is there a more explicit name we can give here?

Reply:
or alternatively, can this just take a boolean as a parameter indicating if its for a worker or not